### ISSUE TITLE
How to search for optimal hyperparameters with TimeGAN?

### ISSUE THREAD

**Issue by rdvhoorn** [jsyoon0823/TimeGAN#67]
Hi Dr. Yoon,

I was wondering what your hyperparameter optimization process looked like for this research? GANs are notoriously hard to fine-tune, and Time-GAN adds additional complexity, which does not make it any easier. 

Did you use some standard optimization process like bayesian optimization? If so, what kind of metrics / combination of metrics did you optimize for? 
Did you do it by hand? If so, what was the approach?

**Comment by jsyoon0823** [comment]
Thanks for your question.

Yes. We also know that it is hard to optimize the hyper-parameters of GAN model.
In our case, we use the fidelity metrics (the realisticity of the synthetic data) as the criteria of the hyper-parameter tuning.
- More specifically, if the performance of trained model (trained on the synthetic data) works well on the real test set, we treat those hyper-parameter sets as high performing combinations.

We use greedy search for finding the hyper-parameters.

Thanks!

**Comment by rdvhoorn** [comment]
Thanks for the response. I'd like to ask a follow-up question:

1. What fidelity metric did you use? In the paper, you mentioned a discriminative score. If you greedily optimize for a discriminative score, wouldn't you just get to the first set of parameters which mode-collapses? If it mode-collapses, you will get very good results for such a discriminative score but the model would be of very bad quality.
2. In the greedy hyper-parameter search, what parameters did you tune? Did you have different learning rates for the different modules (generator, discriminator etc.) separately? Did you have a different number of hidden nodes for the different modules? Did you try different optimizers? 

Thanks in advance!
